src/files-and-videos/videos-page/data/thunks.js

```diff
@@ -103,42 +101,47 @@ export function deleteVideoFile(courseId, id) {
 export function addVideoFile(courseId, file, videoIds) {
   return async (dispatch) => {
     dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.IN_PROGRESS }));
-
     try {
-      const { files } = await addVideo(courseId, file);
-      const { edxVideoId, uploadUrl } = files[0];
-      const errors = await uploadVideo(
-        courseId,
-        uploadUrl,
-        file,
-        edxVideoId,
-      );
-      const { videos } = await fetchVideoList(courseId);
-      const newVideos = videos.filter(video => !videoIds.includes(video.edxVideoId));
-      const parsedVideos = updateFileValues(newVideos, true);
-      dispatch(addModels({
-        modelType: 'videos',
-        models: parsedVideos,
-      }));
-      dispatch(addVideoSuccess({
-        videoId: edxVideoId,
-      }));
-      dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.SUCCESSFUL }));
-      if (!isEmpty(errors)) {
-        errors.forEach(error => {
-          dispatch(updateErrors({ error: 'add', message: error }));
-        });
+      const createUrlResponse = await addVideo(courseId, file);
+      // eslint-disable-next-line
+      console.log(`Post Response: ${createUrlResponse}`);
+      if (createUrlResponse.status < 200 && createUrlResponse.status >= 300) {
+        dispatch(updateErrors({ error: 'add', message: `Failed to add ${file.name}.` }));
+        dispatch(updateEditStatus({ editType: 'add', status: RequestStatus.FAILED }));
+        return;
```

Comment:
Can this be moved to a separate function to help improve the readability of this function?